src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -164,16 +162,32 @@ public async Task<TCommandResult> ExecuteAsync(IConnection connection, Cancellat
             }
             catch (Exception exception)
             {
-                if (exception is MongoException mongoException && ShouldAddTransientTransactionError(mongoException))
-                {
-                    mongoException.AddErrorLabel("TransientTransactionError");
-                }
+                AddErrorLabelIfRequired(exception, connection.Description.ServerVersion);
+
                 TransactionHelper.UnpinServerIfNeededOnCommandException(_session, exception);
                 throw;
             }
         }
 
         // private methods
+        private void AddErrorLabelIfRequired(Exception exception, SemanticVersion serverVersion)
+        {
+            if (exception is MongoException mongoException)
+            {
+                if (ShouldAddTransientTransactionError(mongoException))
+                {
+                    mongoException.AddErrorLabel("TransientTransactionError");
+                }
+
+                if ((exception is MongoConnectionException) || // network error
```

Comment:
According to the ticket, this condition can be rewritten in this way:

* All network errors, no matter the server version, add the error label to the exception
* 4.4+ do not add any error labels to command response
* <4.4 Add the errorLabel if it meets the spec'd requirements